### ISSUE TITLE
Duplicated documents after multiple addAll

### ISSUE THREAD

**Issue by yuklai** [lucaong/minisearch#111]
My understanding is that the default 'id' key is used as the unique key, but I still noticed that after mulitple addAll calls, the documents are duplicated as a result of the search. Am I doing anything wrong?

```
    this.search = new MiniSearch<BrokerProfileReturnType>({
      fields: ['id', 'name'], // fields to index for full-text search
      storeFields: ['id', 'name', 'specialNumber1', 'specialNumber2', 'address.street', 'address.city', 'address.state', 'address.zipcode', 'address.countryCode'], // fields to return with search results
      searchOptions: {
        prefix: true,
        fuzzy: true,
      },
      extractField: (document, fieldName) => {
        // If field name is 'pubYear', extract just the year from 'pubDate'
        if (fieldName === 'specialNumber1') {
          return _.find(document.legalIdentifiers, (id) => id.name === 'SpecialNumber1');
        } if (fieldName === 'specialNumber2') {
          return _.find(document.legalIdentifiers, (id) => id.name === 'SpecialNumber2');
        }
    
        // Access nested fields
        return fieldName.split('.').reduce((doc, key) => doc && doc[key], document)
      }
    });
```

**Comment by lucaong** [comment]
Hi @yuklai ,
Yes, the `id` key uniquely identifies a document. However, adding the same document multiple times (with `add` or with `addAll`) will lead to duplicated results.

If a document changes, in order to “reindex” it one must first delete the old one, then add the new. That’s because of the way inverted indexes work: it’s not generally possible to change a document in the inverted index “in place”.

I hope this clarifies your doubt.

**Comment by lucaong** [comment]
I am closing this issue, but feel free to reply if necessary.

**Comment by lucaong** [comment]
@yuklai I revive this thread to announce that the newly released `v6.0.0-beta.1` (which will soon be `v6.0.0`) contains changes that help with this issue:

  - The methods `add` and `addAll` now throw an error in case of duplicate ID
  - The new method `has` can be used to check if a document with a certain ID already exists
  - The new method `replace` can be used to replace an existing document with a new version

See [the documentation](https://lucaong.github.io/minisearch/classes/_minisearch_.minisearch.html) for more information.
